src/components/Grid/GridTable.tsx

```diff
@@ -0,0 +1,131 @@
+     */
+    progress: null | number;
+}
+
+export interface GridTableProps<TData>
+    extends Omit<
+            GridOptions<TData>,
+            | 'overlayLoadingTemplate'
+            | 'loadingOverlayComponent'
+            | 'loadingOverlayComponentParams'
+        >,
+        Partial<GridProgressProps> {}
+
+/*
+ * Restore lost generics from `forwardRef()`<br/>
+ * https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
+ */
+type ForwardRef<Props, Ref> = typeof forwardRef<Props, Ref>;
+type ForwardRefComponent<Props, Ref> = ReturnType<ForwardRef<Props, Ref>>;
+interface GridTableWithRef
+    extends FunctionComponent<PropsWithChildren<GridTableProps<unknown>>> {
+    <TData, TContext extends {}>(
+        props: PropsWithoutRef<PropsWithChildren<GridTableProps<TData>>> &
+            RefAttributes<AgGridRef<TData, TContext>>
+    ): ReturnType<
+        ForwardRefComponent<GridTableProps<TData>, AgGridRef<TData, TContext>>
+    >;
+}
+
+/**
+ * Common part for a Grid with toolbar
+ * @param props
+ */
+export const GridTable: GridTableWithRef = forwardRef(function AgGridToolbar<
+    TData,
+    TContext extends {} = {}
+>(
+    props: PropsWithChildren<GridTableProps<TData>>,
+    gridRef: ForwardedRef<AgGridRef<TData, TContext>>
+): ReactNode {
+    const { children: toolbarContent, progress, ...agGridProps } = props;
+    const columnTypes = useColumnTypes<TData>();
+    return (
+        <Grid
+            container
+            direction="column"
+            justifyContent="flex-start"
+            alignItems="stretch"
+        >
+            <Grid xs="auto">
+                <AppBar position="static">
+                    <Toolbar
+                        variant="dense"
+                        sx={{
+                            '& > *': {
+                                marginRight: '0.2em',
+                                '&:last-child': {
+                                    marginRight: 0,
+                                },
+                            },
+                        }}
+                    >
+                        {/*TODO button reset grid filter/sort/column-hide/rows-selection ...*/}
+                        {/*<Divider orientation="vertical" variant="middle" flexItem />*/}
+                        {toolbarContent}
+                        <Box sx={{ flexGrow: 1 }} />
+                    </Toolbar>
+                </AppBar>
+            </Grid>
+            <Grid item xs="auto">
+                <GridProgress progress={progress ?? null} />
+            </Grid>
+            <Grid item xs>
+                <AgGrid<TData, TContext>
+                    columnTypes={columnTypes}
+                    {...agGridProps}
+                    ref={gridRef}
+                />
+            </Grid>
+        </Grid>
+    );
+});
+export default GridTable;
+
+const GridProgress: FunctionComponent<GridProgressProps> = (props, context) => {
```

Comment:
What's the use of this ? Do we need it at this stage ? Remove ?

Reply:
It's used for the user feedback that the modification (add or delete) is being apply. because else you have no idea what's the application is doing.